pkg/vere/io/ames.c

```diff
@@ -2060,6 +2060,50 @@ _ames_recv_cb(uv_udp_t*        wax_u,
   }
 }
 
+static void
+_mdns_dear_bail(u3_ovum* egg_u, u3_noun lud)
+{
+  u3l_log("mdns: %%dear failure;");
+  u3z(lud);
+  u3_ovum_free(egg_u);
+}
+
+/* _ames_put_dear(): send lane to arvo after hearing mdns response
+*/
+static void
+_ames_put_dear(char* ship, unsigned long s_addr, uint16_t port, void* context)
+{
+  u3_ames* sam_u = (u3_ames*)context;
+
+  u3_lane lan;
+  lan.pip_w = ntohl(s_addr);
+  lan.por_s = ntohs(port);
+
+  u3_noun shp_u = u3dc("slaw", c3__p, u3i_string(ship));
+
+  if (u3_nul == shp_u) {
+    u3l_log("ames: strange ship from mdns: %s", ship);
+    return;
+  }
+
+  u3_noun our = u3i_chubs(2, sam_u->pir_u->who_d);
+  if (our == u3t(shp_u)) {
+    u3z(our);
+    return;
+  }
+
+  u3z(our);
+
+  u3_noun wir = u3nc(c3__ames, u3_nul);
+  u3_noun cad = u3nt(c3__dear, u3k(u3t(shp_u)), u3nc(c3n, u3_ames_encode_lane(lan)));
+
+  u3_auto_peer(
+               u3_auto_plan(&sam_u->car_u,
+                            u3_ovum_init(0, c3__a, wir, cad)),
+               0, 0, _mdns_dear_bail);
```

Comment:
We don't need to track or handle failures with these events.

Reply:
I get real ugly crashes on arvo versions that don't support `%dear` unless I handle the bail myself.